sdk/identity/Azure.Identity/src/AzurePowerShellCredential.cs

```diff
@@ -198,11 +204,35 @@ private void GetFileNameAndArguments(string resource, out string fileName, out s
 
         private static AccessToken DeserializeOutput(string output)
         {
-            using JsonDocument document = JsonDocument.Parse(output);
+            XDocument document = XDocument.Parse(output);
+            string accessToken = null;
+            DateTimeOffset expiresOn = default;
+
+            if (document?.Root == null)
+            {
+                throw new CredentialUnavailableException("Error parsing token response.");
+            }
 
-            JsonElement root = document.RootElement;
-            string accessToken = root.GetProperty("Token").GetString();
-            DateTimeOffset expiresOn = DateTimeOffset.ParseExact(root.GetProperty("ExpiresOn").GetString(), "yyyy-MM-ddTHH:mm:sszzz", CultureInfo.InvariantCulture, DateTimeStyles.AdjustToUniversal | DateTimeStyles.AssumeLocal);
+            foreach (var e in document.Root.Elements())
+            {
+                switch (e.Attribute("Name")?.Value)
+                {
+                    case "Token":
+                        accessToken = e.Value;
+                        break;
+
+                    case "ExpiresOn":
+                        expiresOn = DateTimeOffset.Parse(e.Value, CultureInfo.InvariantCulture).ToUniversalTime();
```

Comment:
Do we know if the ExpiresOn is going to be in local time or UTC time? If local does it contain a timezone offset? If not we should specify a `DateTimeStyle` as we had to do in the `AzureCliCredential` [here](https://github.com/Azure/azure-sdk-for-net/blob/dc095fd0d8ad0afd44bfdc394c84e3dcf98438e8/sdk/identity/Azure.Identity/src/AzureCliCredential.cs#L182)

Reply:
It appear to be UTC but with an offset also, so the default Parse seems to do the right thing. Do you think we should still try to ParseExact?

`ExpiresOn : 4/21/2021 6:51:50 PM +00:00`

Reply:
No, `ParseExact` shouldn't be necessary. We needed that for the CLI because they have a weird format, I was just checking if we need to specify `DateTimeStyle`, but if the output always contains an offset it's not necessary.